src/basic_server_config.cpp

```diff
@@ -44,68 +36,50 @@ bool BasicServerConfig::InitPortNumber(NginxConfig* config) {
   return true;
 }
 
-bool BasicServerConfig::InitEchoPath(NginxConfig* config) {
-  echo_path_ = GetStatementValue(config, ECHO_KEY);
-  return true;
-}
-
-bool BasicServerConfig::InitRootUrlPaths(NginxConfig* config) {
+bool BasicServerConfig::InitRequestHandlers(NginxConfig* config) {
   std::vector<std::shared_ptr<NginxConfigStatement>> location_matches = 
     FilterStatements(config, LOCATION_OBJ);
   
-  if (location_matches.size() == 0) {
-    return false;
-  }
-  
   for (const auto& statement : location_matches) {
-    if (statement->tokens_.size() >= 2) {
+    if (statement->tokens_.size() >= 3) {
       NginxConfig* location_map_block = statement->child_block_.get();
 
-      std::string user_url = statement->tokens_[1];
-      std::string host_url = GetStatementValue(location_map_block, ROOT_KEY);
+      std::string uri = statement->tokens_[1];
+      std::string handler_id = statement->tokens_[2];
+      std::string root_dir = GetStatementValue(location_map_block, ROOT_KEY);
```

Comment:
Root dir is not a required parameter. Shouldn't be extracting it here, nor passing it into BuildHandlerForUri. You do, however, need to pass in the child block to the BuildHandlerForUri function so that RequestHandler subclasses can extract any information (such as a root directory) that they need.